lib/path.dart

```diff
@@ -361,29 +360,39 @@ int hash(String path) => context.hash(path);
 
 /// Removes a trailing extension from the last part of [path].
 ///
-///     withoutExtension('path/to/foo.dart'); // -> 'path/to/foo'
+///     p.withoutExtension('path/to/foo.dart'); // -> 'path/to/foo'
 String withoutExtension(String path) => context.withoutExtension(path);
 
+/// Returns [path] with the trailing extension changed to [extension].
+///
+/// If [path] doesn't have a trailing extension, this just adds [extension] to
+/// the end.
```

Comment:
Given that it adds the extension if there is none, "change" may not be the best word. How about `setExtension`? That's shorter and makes it clearer that the original extension isn't relevant. Or maybe `withExtension`?

Actually `withExtension` is better since it matches `withoutExtension`.

Reply:
I'd expect `withExtension()` to be the exact dual of `withoutExtension()`... that is, I'd expect `withoutExtension(withExtension(path, extension)) == path`, which isn't the case if it's replacing an existing extension. I like `setExtension()`, though.